updater/internal/install/service_windows.go

```diff
@@ -0,0 +1,276 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+
+package install
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"golang.org/x/sys/windows/registry"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"github.com/kballard/go-shellquote"
+)
+
+const (
+	defaultProductName           = "observIQ Distro for OpenTelemetry Collector"
+	defaultServiceName           = "observiq-otel-collector"
+	uninstallServicePollInterval = 50 * time.Millisecond
+	serviceNotExistErrStr        = "The specified service does not exist as an installed service."
+)
+
+// newService returns an instance of the Service interface for managing the observiq-otel-collector service on the current OS.
+func newService(latestPath string) Service {
+	return &windowsService{
+		newServiceFilePath: filepath.Join(latestPath, "install", "windows_service.json"),
+		serviceName:        defaultServiceName,
+		productName:        defaultProductName,
+	}
+}
+
+type windowsService struct {
+	// newServiceFilePath is the file path to the new unit file
+	newServiceFilePath string
+	// serviceName is the name of the service
+	serviceName string
+	// productName is the name of the installed product
+	productName string
+}
+
+// Start the service
+func (w windowsService) Start() error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return fmt.Errorf("failed to connect to service manager: %w", err)
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(w.serviceName)
+	if err != nil {
+		return fmt.Errorf("failed to open service: %w", err)
+	}
+	defer s.Close()
+
+	if err := s.Start(); err != nil {
+		return fmt.Errorf("failed to start service: %w", err)
+	}
+
+	return nil
+}
+
+// Stop the service
+func (w windowsService) Stop() error {
+	m, err := mgr.Connect()
+	if err != nil {
+		return fmt.Errorf("failed to connect to service manager: %w", err)
+	}
+	defer m.Disconnect()
+
+	s, err := m.OpenService(w.serviceName)
+	if err != nil {
+		return fmt.Errorf("failed to open service: %w", err)
+	}
+	defer s.Close()
+
+	if _, err := s.Control(svc.Stop); err != nil {
+		return fmt.Errorf("failed to start service: %w", err)
+	}
+
+	return nil
+}
+
+// Installs the service
+func (w windowsService) Install() error {
+	wsc, err := readWindowsServiceConfig(w.newServiceFilePath)
```

Comment:
Can we add comments in here to explain steps more? This is not as straight forward as macOS and Linux